test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpFnTest.java

```diff
@@ -0,0 +1,222 @@
+                {"count(//Customer/node())",
+                        ID_ATTRIBUTES + CUSTOMERS * CUSTOMER_ELEMENTS},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expression for position function.
+     * Data columns:
+     *  see parameters of the test "testPositionFn"
+     */
+    @DataProvider(name = "positionExpTestCases")
+    public Object[][] getPositionExp() {
+        return new Object[][]{
+                {"//Customer[position()=1]", "Customer_x1"},
+                {"//Customer[position()=last()]", "Customer_x3"},
+                {"//Customer[position()>1 and position()<last()]",
+                        "Customer_x2"},
+                {"//Customer[position() mod 2 =0]", "Customer_x2"},
+                {"//Customer[last()]", "Customer_x3"},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expression for name and local name function.
+     * Data columns:
+     *  see parameters of the test "testNameFn"
+     */
+    @DataProvider(name = "nameExpTestCases")
+    public Object[][] getNameExp() {
+        return new Object[][]{
+                {"local-name(//Customer)", "Customer"},
+                {"local-name(//foo:Customer)", "Customer"},
+                {"local-name(//Customer/@id)", "id"},
+                {"local-name(//foo:Customer/@foo:id)", "id"},
+                {"local-name(//*[local-name()='Customer'])", "Customer"},
+                {"namespace-uri(.)", ""},
+                {"namespace-uri(//Customers)", ""},
+                {"namespace-uri(//Customer)", ""},
+                {"namespace-uri(//foo:Customer)", "foo"},
+                {"namespace-uri(//@id)", ""},
+                {"namespace-uri(//@foo:id)", "foo"},
+                {"name(//*[namespace-uri()=\"foo\"])", "foo:Customer"},
+                {"name(//Customer)", "Customer"},
+                {"name(//foo:Customer)", "foo:Customer"},
+                {"name(//Customer/@id)", "id"},
+                {"name(//foo:Customer/@foo:id)", "foo:id"},
+                {"name(//*[name()='foo:Customer'])", "foo:Customer"},
+        };
+    }
+
+    /**
+     * This test evaluates XPath expressions of id function and checks against
+     * the expected result.
+     *
+     * @param exp      XPath expression
+     * @param expected expected result
+     * @throws Exception
+     */
+    @Test(dataProvider = "idExpTestCases")
+    void testIdFn(String exp, String expected) throws Exception {
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        Node node = xPath.evaluateExpression(exp, doc, Node.class);
+        Node node2 = (Node) xPath.evaluate(exp, doc, XPathConstants.NODE);
+
+        Assert.assertEquals(node.getNodeName() + "_" +
+                        node.getAttributes().item(0).getNodeValue()
+                , expected);
+        Assert.assertEquals(node2, node);
+    }
+
+    /**
+     * This test evaluates XPath expressions of count function and checks
+     * against the expected result.
+     *
```

Comment:
Verifies that the result of evaluating the count function matches the expected result.